packages/react/menu/src/Menu.tsx

```diff
@@ -710,8 +743,31 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
         data-state={getOpenState(context.open)}
         {...props}
         ref={composeRefs(forwardedRef, context.onTriggerChange)}
-        onMouseMove={composeEventHandlers(props.onMouseMove, () => {
-          if (!props.disabled) context.onOpenChange(true);
+        onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
+          if (contentContext.isMouseMovingToSubmenu(event)) return;
+          if (!props.disabled && !context.open) {
+            openTimersRef.current.push(window.setTimeout(() => context.onOpenChange(true), 100));
+          }
+        })}
+        onMouseLeave={composeEventHandlers(props.onMouseLeave, (event) => {
+          openTimersRef.current.map(window.clearTimeout);
+          openTimersRef.current = [];
+
+          const contentRect = context.content?.getBoundingClientRect();
```

Comment:
I'm wondering if we can store the bounding client rect in a ref somewhere when the content opens instead of every time the mouse leaves. 

Atm, this looks like we'll be getting a new bounding rect every time user does this:

https://user-images.githubusercontent.com/175330/120836251-42e70980-c55d-11eb-8833-b7b2b81d0844.mp4

Is that correct? Seems like unnecessary reflow.

Reply:
We could cache it, when do we clear the cache? on unmount? what happens if new items get added dynamically? Maybe we don't worry about it?

Reply:
I dunno, you asked me to store this stuff in `ScrollArea` so I thought we'd be consistent. Given the perf issues we had on the design-system page lately, it seems this would be pretty heavy there and we're best off trying to avoid reflows where possible.

> what happens if new items get added dynamically

Could maybe worry about that when someone reports an issue?

Reply:
Happy to go with simple caching for now for the life of the open submenu.